gtsfm/evaluation/metrics_report.py

```diff
@@ -0,0 +1,205 @@
+            histogram = metric_value[metrics.SUMMARY_KEY]["histogram"]
+            fig.add_trace(
+                go.Bar(x=list(histogram.keys()), y=list(histogram.values()), name=metric_name), row=row, col=col
+            )
+        elif "quartiles" in metric_value[metrics.SUMMARY_KEY]:
+            # If all values are available, use them to create box plot.
+            if metrics.FULL_DATA_KEY in metric_value:
+                fig.add_trace(go.Box(y=metric_value[metrics.FULL_DATA_KEY], name=metric_name), row=row, col=col)
+            # Else use summary to create box plot.
+            else:
+                quartiles = metric_value[metrics.SUMMARY_KEY]["quartiles"]
+                fig.add_trace(
+                    go.Box(
+                        q1=[quartiles["q1"]],
+                        median=[quartiles["q2"]],
+                        q3=[quartiles["q3"]],
+                        lowerfence=[quartiles["q0"]],
+                        upperfence=[quartiles["q4"]],
+                        name=metric_name,
+                    ),
+                    row=row,
+                    col=col,
+                )
+
+    # Return the figure converted to HTML.
+    return fig.to_html(full_html=False, include_plotlyjs="cdn")
+
+
+def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]:
+    """Gets the tables and plots for individual metrics in a metrics group.
+
+    All scalar metrics are reported in the table.
+    Metrics of 1-D distributions have an entry in the table for the mean,
+    and a histogram or box plot as per their property.
+
+    Args:
+        metrics_group: A GtsfmMetricsGroup for any gtsfm module.
+
+    Returns:
+        A tuple of table and plotly figures as HTML code.
+    """
+    scalar_metrics = {}
+    metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
+
+    # Separate the scalar metrics.
+    for metric, value in metrics_dict.items():
+        if isinstance(value, dict):
+            # Metrics with a dict representation must contain a summary.
+            if not metrics.SUMMARY_KEY in value:
+                raise ValueError("Metric {metric} does not contain a summary.")
+            # Add a scalar metric for mean of 1D distributions.
+            scalar_metrics["mean_" + metric] = value[metrics.SUMMARY_KEY]["mean"]
+        else:
+            scalar_metrics[metric] = value
+    table = create_table_for_scalar_metrics(scalar_metrics)
+    plots_fig = create_plots_for_distributions(metrics_dict)
+    return table, plots_fig
+
+
+def get_html_metric_heading(metric_name: str) -> str:
+    """Helper to get the HTML heading for a metric name.
+
+    This converts a "metric_name" to "Metric Name" (for readability) and adds some style.
+
+    Returns:
+        HTML for heading as a string.
+    """
+    metric_name = get_readable_metric_name(metric_name)
+    metric_html = f'<p style="font-size:25px;font-family:Arial">{metric_name}</p>'
```

Comment:
sorry for being an HTML noob, but can we not have an accompanying css sheet and save on the styling code here?

Reply:
Maybe.. but me being even noob-er did the minimum to make it look not bad. I dont quite know how this is done, but maybe we can add the TODO to improve the HTML here. Whoever has some HTML experience can take it up, or maybe I can do it someday.